smarts/core/tests/test_renderers.py

```diff
@@ -103,25 +140,38 @@ def test_renderer(self):
         self._rdr.remove_vehicle_node(self._vid)
 
 
-def test_multiple_renderers(scenario):
+def test_multiple_renderers(scenario, request):
+    renderer_debug_mode = request.config.getoption("--renderer-debug-mode")
     num_renderers = 3
-    rts = [RenderThread(r, scenario) for r in range(num_renderers)]
+    rts = [RenderThread(r, scenario, renderer_debug_mode) for r in range(num_renderers)]
     for rt in rts:
         rt.start()
     for rt in rts:
         rt.join()
 
 
-def test_optional_renderer(smarts, scenario):
-    smarts.reset(scenario)
+def test_optional_renderer(smarts: SMARTS, scenario):
     assert not smarts.is_rendering
-    for _ in range(10):
-        smarts.step({})
-
     renderer = smarts.renderer
     if not renderer:
         raise RendererException.required_to("run test_renderer.py")
 
+    smarts._renderer = None
```

Comment:
I think we also need to check that `smarts._renderer` defaults to `None` if no camera sensors are attached.

Reply:
That is done in `test_no_renderer(..)`.